Automattic/wp-calypso
# JP Settings: Rename and fix action types

This PR:

* Removes obsolete action types (`JETPACK_SETTINGS_REGENERATE_POST_BY_EMAIL_*`, `JETPACK_SETTINGS_REQUEST_*`)
* Fixes the Jetpack Modules reducer
  * This was listening to `JETPACK_SETTINGS_RECEIVE` and `JETPACK_SETTINGS_UPDATE_SUCCESS`, which as of #23797 were no longer fired.
  * Their respective replacements are `JETPACK_ONBOARDING_SETTINGS_UPDATE`  and `JETPACK_ONBOARDING_SETTING_SAVE_SUCCESS` (see table below)
* Finally, after removing all legacy `JETPACK_SETTINGS_*` action types, we `s/JETPACK_ONBOARDING_SETTINGS/JETPACK_SETTINGS/`.

| Before #23797               | After                     | What it does                     |
|-----------------------------|---------------------------|----------------------------------|
| `fetchSettings()`           | `requestJetpackSettings()` | Fetch Settings from Network      |
| `updateSettings()`          | `saveSettings()`           | POST updated settings to Network |
| `JETPACK_SETTINGS_UPDATE_*` | `JETPACK_SETTINGS_SAVE_*`  | POST updated settings to Network |
| `JETPACK_SETTINGS_RECEIVE`  | `JETPACK_SETTINGS_UPDATE`  | Update settings in Redux state   |

### Testing Instructions

Since we're renaming action types, a bit of smoke testing that covers all affected action types should quickly reveal if anything's broken.

* Test JPO
* Test JP Settings
  * Try updating some settings
  * Try regenerating the post-by-email address
  * Try updating the Akismet API key to something invalid
* Verify that the JP Modules Redux state (`state.jetpack.modules`) is correctly updated when a setting is enabled. (I'm not 100% how to test this):
  * Get a fresh JP site that doesn't have all modules activated
  * Enable a few JP settings
  * Check Redux action log: the `JETPACK_SETTINGS_SAVE_SUCCESS` and/or `JETPACK_SETTINGS_UPDATE` actions should flip the corresponding module's activation state to `true`.
